commit 997a517475cfd7fa0191fa06421e5d042fef09ba
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2019-07-17T16:48:20+00:00 (3 months ago)
    
    OS-6168 Attempt to first unmount "cores" before renaming the delegated dataset during reprovisioning

diff --git a/src/vm/node_modules/VM.js b/src/vm/node_modules/VM.js
index 7d48bbd5..28b33dfd 100644
--- a/src/vm/node_modules/VM.js
+++ b/src/vm/node_modules/VM.js
@@ -10886,25 +10886,6 @@ exports.reprovision = function (uuid, payload, options, callback)
                 }
                 cb(err);
             });
-        }, function (cb) {
-            // if we have a delegated dataset, rename zones/<uuid>/data
-            //     -> zones/<uuid>-reprovisioning-data
-            if (!vmobj.hasOwnProperty('datasets')
-                || vmobj.datasets.length === 0) {
-
-                cb();
-                return;
-            }
-            zfs(['rename', '-f', vmobj.datasets[0], vmobj.zfs_filesystem
-                + '-reprovisioning-data'], log, function (err, fds) {
-
-                if (err) {
-                    log.error({err: err, stdout: fds.stdout,
-                        stderr: fds.stderr}, 'Unable to (temporarily) rename '
-                        + vmobj.datasets[0]);
-                }
-                cb(err);
-            });
         }, function (cb) {
             // unmount <zonepath>/cores so dataset is not busy
             zfs(['umount', vmobj.zonepath + '/cores'], log,
@@ -10924,6 +10905,25 @@ exports.reprovision = function (uuid, payload, options, callback)
                 }
                 cb(err);
             });
+        }, function (cb) {
+            // if we have a delegated dataset, rename zones/<uuid>/data
+            //     -> zones/<uuid>-reprovisioning-data
+            if (!vmobj.hasOwnProperty('datasets')
+                || vmobj.datasets.length === 0) {
+
+                cb();
+                return;
+            }
+            zfs(['rename', '-f', vmobj.datasets[0], vmobj.zfs_filesystem
+                + '-reprovisioning-data'], log, function (err, fds) {
+
+                if (err) {
+                    log.error({err: err, stdout: fds.stdout,
+                        stderr: fds.stderr}, 'Unable to (temporarily) rename '
+                        + vmobj.datasets[0]);
+                }
+                cb(err);
+            });
         }, function (cb) {
             // rename <zfs_filesystem> dataset out of the way
             var cancelFn;
